cmd/tracee-ebpf/internal/printer/printer.go

```diff
@@ -198,16 +211,23 @@ func (p templateEventPrinter) Close() {
 }
 
 type jsonEventPrinter struct {
-	out io.WriteCloser
-	err io.WriteCloser
+	out      io.WriteCloser
+	err      io.WriteCloser
+	protocol bool
 }
 
 func (p jsonEventPrinter) Init() error { return nil }
 
 func (p jsonEventPrinter) Preamble() {}
 
 func (p jsonEventPrinter) Print(event trace.Event) {
-	eBytes, err := json.Marshal(event)
+	var err error
+	var eBytes []byte
+	if p.protocol {
+		eBytes, err = json.Marshal(event.ToProtocol())
```

Comment:
This still looks to me like a workaround. Why do we perform conversion from a specific to generic event type so late at print time? Shouldn't we update the eventPrinter interface and adapt the upstream code as well? In particular, send protocol.events to the `cfg.ChanEvents`? 

```go
type eventPrinter interface {
	// Print prints a single event
	Print(event protocol.Event)

	// ... other methods trimmed ... //
```

Now passing generic event to printer implementors, will allows us to keep GOB or JSON printers intact (by just taking the Payload), but at the same time will allow us to enhance them with event metadata if we like. And still I think that the protocol flag passed to tracee-ebpf in unnecessary. We have tens of flags already.

Reply:
The problem with this way is that json or gob printers won't just print out the payload, we would have to explicitly tell them to do so inside their implementations (since the event would be passed with metadata), basically just doing the opposite of the protocol flag - so we would still have to define a way to allow them to be printed without the headers.

Also by passing `protocol.Event` instead of `trace.Event` how will it affect all the other printers? They would have to check if the payload is actually a `trace.Event` and only then print (for example the table printer), are we okay with that?

About having `cfg.ChanEvents` send the events as protocol already, that would mean any other consumer would have to either support our message format, or convert to their own protocol with a middleware, not so bad, but if we only output a single event it seems redundant to me. Maybe if we started outputting different payloads it would make sense instead of emitting from multiple channels.

Reply:
I've added a commit so that Print takes a `protocol.Event` instead of a `trace.Event` please take a look. I still have reservations about outputting `protocol.Event` directly from tracee-ebpf `ChanEvents`, I suggest we take it to this week's sync.